src/aggregate/aggregate_request.c

```diff
@@ -1192,6 +1288,23 @@ int AREQ_ApplyContext(AREQ *req, RedisSearchCtx *sctx, QueryError *status) {
     return REDISMODULE_ERR;
   }
 
+  if (req->parsedVectorQuery) {
+    QAST_ValidationFlags prevFlags = ast->validationFlags;
+    ast->validationFlags |= QAST_HYBRID_VSIM_FILTER_CLAUSE;
+    if (QAST_CheckIsValid(ast, AREQ_SearchCtx(req)->spec, opts, status) != REDISMODULE_OK) {
```

Comment:
Let's not do this (almost) twice.
If we can move the check below to before this, best. If not, let's mark this vector as ok for the validation